functions/post_data.go

```diff
@@ -0,0 +1,147 @@
+package functions
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+)
+
+const timeoutSeconds = 10 // the max number of seconds to wait for a response
+var errURLMissing = errors.New("error calling PostData; URL is required")
+var errURLEmpty = errors.New("error calling PostData; URL is empty")
+
+// PostData sends an HTTP POST request to a given URL with the data provided,
+// returning either the whole response body or a specified JSON element.
+func PostData(existingHeaders map[string]string, args []string) (string, error) {
```

Comment:
I decided to keep it specific to url-encoded forms, since we might want to do others things differently for different content types. So it's `PostFormUrlencoded` now.